### PR TITLE
added three constructors to code_function_callt

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: src/util/std_code.h [hunk 1]
```diff
@@ -872,6 +872,36 @@ class code_function_callt:public codet
     op2().id(ID_arguments);
   }
 
+  explicit code_function_callt(const exprt &_function) : codet(ID_function_call)
```
Comment (peterschrammel) [inline src/util/std_code.h:876]: Mark the no-argument constructor as deprecated.

**Review by peterschrammel — APPROVED** [review]

File: src/util/std_code.h (hunk 1 above)
Comment (kroening) [reply src/util/std_code.h:876]: done

**Comment by thk123** [comment]
TG bump (CI running): diffblue/test-gen#2204

File: src/ansi-c/ansi_c_entry_point.cpp [hunk 2]
```diff
@@ -213,19 +213,17 @@ bool generate_ansi_c_start_function(
       return true;
     }
 
-    code_function_callt call_init;
-    call_init.lhs().make_nil();
+    code_function_callt call_init(
+      nil_exprt(), init_it->second.symbol_expr(), {});
```
Comment (tautschnig) [inline src/ansi-c/ansi_c_entry_point.cpp]: I think `call_init(init_it->second.symbol_expr())` would do?

**Review by tautschnig — APPROVED** [review]

File: src/ansi-c/ansi_c_entry_point.cpp (hunk 2 above)
Comment (kroening) [reply src/ansi-c/ansi_c_entry_point.cpp]: done

**Comment by tautschnig** [comment]
To fix up CI this requires a rebase once #2863 has been merged.

**Pull request merged**